added checks about the status of a video

SMN321 opened the pull request:
added a check if the video is uploaded and a check if the video is private

SMN321 committed:
added checks about the status of a video

added a check if the video is uploaded and a check if the video is private

ottojo commented on intercept.go:
```diff
@@ -115,14 +115,24 @@ func analyzeVideo(videoID string) (reason string, suspicious bool) {
+			suspicious = true
+		}
+		
+		//TODO: Maybe add unlisted videos as well
```
Dafür bitte einfach ein Issue aufmachen anstatt TODO comments einzufügen

ottojo commented on intercept.go:
```diff
@@ -115,14 +115,24 @@ func analyzeVideo(videoID string) (reason string, suspicious bool) {
-
+		
+		if video.Status.UploadStatus != "uploaded" {
+			appendStringComma(&reasons, "Video is not uploaded")
```
Hier fände ich cool wenn der tatsächliche UploadStatus in der Fehlermeldung wäre, z.B.
"Video status is \"" + video.Status.UploadStatus + "\""

https://developers.google.com/youtube/v3/docs/videos#status.uploadStatus

ottojo replied:
reasons -> reason

ottojo commented on intercept.go:
```diff
@@ -115,14 +115,24 @@ func analyzeVideo(videoID string) (reason string, suspicious bool) {
+		
+		//TODO: Maybe add unlisted videos as well
+		if video.Status.PrivacyStatus == "private" {
+			appendStringComma(&reasons, "Video is private")
```
siehe oben

SMN321 committed:
fixed bugs

reasons -> reason
a 'normal' video has "processed" not "uploaded" as UploadStatus
